commit 58cf4d76cf503b7254709dc94dc51c19fa92d685
Author: Mark Emlyn David Thomas <markt@apache.org>
Date:   Tue Sep 20 17:08:41 2011 +0000

    Ensure that when the connector is stopped that a Comet END event is processed to signal to the app and to close the socket
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1173256 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/tomcat/util/net/NioEndpoint.java b/java/org/apache/tomcat/util/net/NioEndpoint.java
index 6a2ae91..e5e59f9 100644
--- a/java/org/apache/tomcat/util/net/NioEndpoint.java
+++ b/java/org/apache/tomcat/util/net/NioEndpoint.java
@@ -1160,7 +1160,7 @@ public class NioEndpoint extends AbstractEndpoint {
             boolean result = true;
             try {
                 if ( close ) {
-                    cancelledKey(sk, SocketStatus.STOP, false);
+                    cancelledKey(sk, SocketStatus.STOP, attachment.comet);
                 } else if ( sk.isValid() && attachment != null ) {
                     attachment.access();//make sure we don't time out valid sockets
                     sk.attach(attachment);//cant remember why this is here
